share/gpodder/extensions/subscription_stats.py

```diff
@@ -0,0 +1,186 @@
+        dayscolumn = Gtk.TreeViewColumn(_('Days'))
+        dayscolumn.pack_start(dayscell, True)
+        dayscolumn.add_attribute(dayscell, 'text', 0)
+        tree.append_column(dayscolumn)
+
+        channelcell = Gtk.CellRendererText()
+        channelcell.set_property('xalign', 0)
+        channelcell.set_property('alignment', Pango.Alignment.LEFT)
+        channelcolumn = Gtk.TreeViewColumn(_('Podcast'))
+        channelcolumn.pack_start(channelcell, True)
+        channelcolumn.add_attribute(channelcell, 'text', 1)
+        tree.append_column(channelcolumn)
+
+        lastcell = Gtk.CellRendererText()
+        lastcell.set_property('xalign', 1)
+        lastcell.set_property('alignment', Pango.Alignment.LEFT)
+        lastcolumn = Gtk.TreeViewColumn(_('Last Updated'))
+        lastcolumn.pack_start(lastcell, True)
+        lastcolumn.add_attribute(lastcell, 'text', 2)
+        tree.append_column(lastcolumn)
+
+        channellastcell = Gtk.CellRendererText()
+        channellastcell.set_property('xalign', 0)
+        channellastcell.set_property('alignment', Pango.Alignment.LEFT)
+        channellastcolumn = Gtk.TreeViewColumn(_('Last Feed Updated'))
+        channellastcolumn.pack_start(channellastcell, True)
+        channellastcolumn.add_attribute(channellastcell, 'text', 3)
+        tree.append_column(channellastcolumn)
+
+        notebook.append_page(scrolled, Gtk.Label('%d %s %s' % (len(channels), type, type)))
+
+    def open_dialog(self):
+        db = self.gpodder.db
+
+        # get all channels
+        channels = []
+        with db.lock:
+            cur = db.cursor()
+            cur.execute('SELECT id, http_last_modified, title, pause_subscription FROM %s' % db.TABLE_PODCAST)
+            while True:
+                row = cur.fetchone()
+                if row is None:
+                    break
+                channels.append(row)
+            cur.close()
+
+        # get average time between episodes per channel
+        now = int(time.time())
+        nr_paused = 0
+        daily = []
+        weekly = []
+        monthly = []
+        yearly = []
+        for channel_id, channel_last, channel_name, paused in channels:
+            if paused:
+                nr_paused += 1
+
+            total = 0
+            nr_episodes = 0
+            prev = now
+            with db.lock:
+                cur = db.cursor()
+                cur.execute('SELECT published FROM %s WHERE podcast_id = %d ORDER BY published DESC LIMIT 25'
+                    % (db.TABLE_EPISODE, channel_id))
+                while True:
+                    row = cur.fetchone()
+                    if row is None:
+                        break
+                    if total == 0:
+                        edate = row[0]
+                    total += (prev - row[0])
+                    nr_episodes += 1
+                    prev = row[0]
+                cur.close()
+
+            average = (total / nr_episodes) / 86400 if nr_episodes > 0 else 0
```

Comment:
Expanding this product to `24 * 60 * 60` might be clearer.

Reply:
ok, trivial change